slurm_ops_manager/singularity.py

```diff
@@ -0,0 +1,56 @@
+"""Singularity Class."""
+
+import logging
+import shlex
+import subprocess
+from pathlib import Path
+
+from ops.framework import Object, StoredState
+from slurm_ops_manager.utils import operating_system
+
+
+logger = logging.getLogger()
+
+
+class Singularity(Object):
+    """Responsible for Singularity operations."""
+
+    _stored = StoredState()
```

Comment:
Although, possibly you need to inherit `Object` to use `StoredState()` .... I'm not sure actually

Reply:
Per the ops examples, that is required.